hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/access/AbstractAccessDelegate.java

```diff
@@ -108,11 +108,12 @@ public boolean putFromLoad(final Object key, final Object value, final long txTi
         return putFromLoad(key, value, txTimestamp, version);
     }
 
-    /**
-     * This is an asynchronous cache access strategy.
-     * NO-OP
-     */
     public void remove(final Object key) throws CacheException {
+        try {
+            cache.remove(key);
+        } catch (HazelcastException e) {
+            throw new CacheException("Operation timeout during remove operation from cache!", e);
+        }
```

Comment:
It was removed very intentionally. `remove` is for when cache's use synchronous eviction. Removals happen a different way with asynchronous eviction (i.e. soft locking).

The main issue that I believe you are running into is that CollectionRegion and the entity cache use the same underlying implementation (i.e. `AbstractAccessDelegate`). In the Ehcache version they are implemented as separate classes.

My changes made the implementation match closer to what is expected and documented on the Hibernate SPI.

This is all off the top of my head so some details may not be entirely true. I'm happy to explain IRL.